public/tools/auto-lighthouse/index.js

```
@@ -6,71 +6,106 @@ const fs = require("fs");
 const Excel = require("exceljs");
 const lighthouse = require("lighthouse");
 const chromeLauncher = require("chrome-launcher");
+const flags = require("flags");
 
-const RESULTS_FOLDER = "./public/scan-results/data";
+flags.defineString('inputFile', null, 'File containing urls to scan.');
+flags.defineString('inputFormat', 'xls', 'Format of inputFile. Either xls or text.');
+flags.defineString('outputPrefix', '', 'Filename prefix to use when writing reports.');
+flags.defineString('ticket', 0, 'If specified, write reports to C4G-$ticket_CD_$i.json.');
+flags.defineBoolean('linux', false, 'Run on linux. Disable Chrome sandbox.');
+flags.defineBoolean('headless', false, 'Use headless chrome.');
+flags.defineBoolean('clobber', false, 'If True, overwrite existing json reports.');
 
-function launchChromeAndRunLighthouse(url, opts, config = null) {
-  return chromeLauncher
-    .launch({ chromeFlags: opts.chromeFlags })
-    .then((chrome) => {
-      opts.port = chrome.port;
-      return lighthouse(url, opts, config).then((results) => {
-        return chrome.kill().then(() => results.report);
-      });
-    });
-}
+const RESULTS_FOLDER = "./public/scan-results/data";
 
 const readWorkbook = async (filepath, resultsPrefix) => {
   const startTime = Date.now();
 
-  console.log("Reading xlsx file...");
-  const workbook = new Excel.Workbook();
-  await workbook.xlsx.readFile(filepath);
-  const worksheet = workbook.getWorksheet(1);
-  // Find the url column
-  var columnHeader = "";
-  var columnNumber = 0;
-  while (columnHeader != "url" && columnNumber<worksheet.columnCount) {
-    columnNumber++;
-    columnHeader = worksheet.getRow(1).getCell(columnNumber).value;
-  }
-  if (columnHeader == null) {
-    console.error('Couldn\'t find the url column');
+  var urlsToAudit = [];
+  var badUrls = [];
+  // TODO(rousik): do something with this badUrls list.
```

I still prefer failing fast on bad URLs, though that's less desirable when you're running on thousands at once. In excel files, these bad URLS are often null, and without recording which line they were found on, it's harder to determine what they were.

---

I have added console.error here. I have expected that fixing the url and then rerunning w/ --noclobber (default) should just insert the missing file. But because we are dropping valid urls into the array the indexing would change so I have changed that as well (I'm adding nulls into the arrays and skipping them during processing.